base/show.jl

```diff
@@ -1347,47 +1371,77 @@ function show_unquoted(io::IO, ex::Expr, indent::Int, prec::Int)
     elseif (nargs == 1 && head in (:return, :const)) ||
                           head in (:local,  :global, :export)
         print(io, head, ' ')
-        show_list(io, args, ", ", indent)
+        show_list(io, args, ", ", indent, 0, quote_level)
 
     elseif head === :macrocall && nargs >= 2
-        # first show the line number argument as a comment
-        if isa(args[2], LineNumberNode) || is_expr(args[2], :line)
-            print(io, args[2], ' ')
-        end
-        # Use the functional syntax unless specifically designated with prec=-1
-        # and hide the line number argument from the argument list
-        if prec >= 0
-            show_call(io, :call, args[1], args[3:end], indent)
+        # handle some special syntax
+        # `a b c`
+        if is_core_macro(args[1], "@cmd")
+            print(io, "`", args[3], "`")
+        # 11111111111111111111, 0xfffffffffffffffff, 1111...many digits...
+        elseif is_core_macro(args[1], "@int128_str") ||
+               is_core_macro(args[1], "@uint128_str") ||
+               is_core_macro(args[1], "@big_str")
+            print(io, args[3])
+        # x"y" and x"y"z
+        elseif isa(args[1], Symbol) &&
+               startswith(string(args[1]::Symbol), "@") &&
+               endswith(string(args[1]::Symbol), "_str")
+            s = repr(args[1])
```

Comment:
`string(::Symbol)` is more efficient in principle than `repr` so you may as well use that if you don't need the `:` prefix. (It looks like it's not as efficient as `String(::Symbol)` right now. Probably for no good reason.)

Reply:
Ok, I've changed it accordingly.